tests/wrappers/test_self_supervised_loss_wrapper.py

```diff
@@ -0,0 +1,190 @@
+import inspect
+import unittest
+
+import numpy as np
+import torch
+
+import pytorch_metric_learning.losses as losses
+from pytorch_metric_learning.wrappers import SelfSupervisedLossWrapper
+
+from .. import TEST_DEVICE, TEST_DTYPES
+from ..zzz_testing_utils.testing_utils import angle_to_coord
+
+
+class TestSelfSupervisedLossWrapper(unittest.TestCase):
+    def test_ssl_wrapper_same(self):
+        """
+        Test for distance ~ 0 when given same input embeddings
+        for distance metrics where this is true.
+
+        Other distance metrics that are in supported_loses() not included here (commented)
+        have caveats that prevent it from reaching 0 values.
+
+        Another thing to note is distance -> 0 is less true the
+            1. smaller the embedding sizes are
+            2. the less embeddings there are to consider.
+        This is why the test here uses 200 embeddings of size 512 and atol=1e-3
+        """
```

Comment:
When ```embeddings``` and ```ref_emb``` are identical, the positive pairs have 0 loss, but the negative pairs can still have non-zero loss right?

Reply:
Ohh right, I see. I guess that's why some of these have not produced non-zero losses. So for loss functions that do produce these negative pair margins, how do you suppose we test whether they're working properly?

Reply:
I don't think there's an easy way to test them. For example the triplet loss with margin 0.05 will have a loss of ```[0 - neg_pair_distance + 0.05]```. You basically have to compare it with the original loss function, which you already do in ```test_ssl_wrapper_all```

Reply:
right, maybe we should just get rid of this test. Seems like its unnecessary and making unjustified assumptions.

Reply:
Ok, makes sense!